# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: browser/components/touchbar/MacTouchBar.js
# Commit: 6f13ec193094
# Full Hash: 6f13ec193094a7c7507edd8df79fdb0e1c985c40
# Author: harry <htwyford@mozilla.com>
# Date: 2019-10-11 21:50:12
# Description:
#   Bug 1588084 - Add null-checking to isUrlbarFocused to avoid crashes on window shutdown. r=mikedeboer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D48966
# ==============================================================================

diff -r f3f4aa35d46e -r 6f13ec193094 browser/components/touchbar/MacTouchBar.js
--- a/browser/components/touchbar/MacTouchBar.js	Thu Oct 10 18:51:45 2019 +0000
+++ b/browser/components/touchbar/MacTouchBar.js	Fri Oct 11 13:35:04 2019 +0000
@@ -293,6 +293,9 @@
   }
 
   get isUrlbarFocused() {
+    if (!TouchBarHelper.window || !TouchBarHelper.window.gURLBar) {
+      return false;
+    }
     return TouchBarHelper.window.gURLBar.focused;
   }
 